backend/api/views/document.py

```diff
@@ -123,74 +68,25 @@ def upload_document(id):
     if message != None:
         return create_response(status=400, message=message)
 
-    if "fileName" not in data:
-        return create_response(status=400, message="No file name provided")
-
-    fileName = data.get("fileName")
-
-    if request.files is None or "file" not in request.files:
-        return create_response(status=400, message="No file provided")
-
-    file = request.files.get("file")
-
-    file_info = upload_file(file, fileName)
-
     doc = Document.query.get(id)
 
-    doc.fileID = file_info["file"].id
-    doc.link = file_info["link"]
-    doc.status = "Pending"
-    doc.fileName = fileName
-
-    db.session.commit()
-
-    return create_response(status=200, message="success")
-
-
-@document.route("/document/new", methods=["POST"])
-def create_new_document():
-    """
-    functionality used to add a new document to database
-    """
-    data = request.form
-
-    if data is None:
-        return create_response(status=400, message="No body provided for new Document")
-    # Each document requires a mandatory userID, status (By Default Missing), and a Document Class
-
-    token = request.headers.get("token")
-    headers = {"Content-type": "application/x-www-form-urlencoded", "token": token}
-
-    message, info = verify_token(token)
-    if message != None:
-        return create_response(status=400, message=message)
-
-    if info == "fp":
-        return create_response(
-            status=400, message="You do not have permission to create new documents!"
-        )
-
-    if "userID" not in data:
-        return create_response(
-            status=400, message="No UserID provided for new Document"
-        )
-    if "status" not in data:
-        return create_response(
-            status=400, message="No Status provided for new Document"
-        )
-    if "docClassID" not in data:
-        return create_response(
-            status=400, message="No Document Class provided for new Document"
-        )
+    if "fileName" in data and request.files is not None and file in request.files:
+        fileName = data.get("fileName")
+        file = request.files.get("file")
+        file_info = upload_file(file, fileName)
+        doc.fileID = file_info["file"].id
+        doc.link = file_info["link"]
+        doc.fileName = fileName
 
-    new_data = Document(data)
+    if "status" in data:
+        doc.status = data.get("status")
 
-    db.session.add(new_data)
     db.session.commit()
+
     return create_response(status=200, message="success")
 
 
-@document.route("/document/create", methods=["POST"])
+@document.route("/documents", methods=["POST"])
```

Comment:
Is there a reason for separating the methods? IIRC multiple methods can be specified in the methods list so it could be joined with the previous GET route, but if it's for clarity then this should be fine

Reply:
uhh, no real reason - I guess it is clearer this way but up to you what u want